### PR TITLE
Fix lazy builder class cast

### PR DESCRIPTION
**Pull request by cwlaird3**
This is a fix for https://github.com/bedatadriven/renjin/issues/96

### CODE REVIEW

**Comment by akbertram** [comment]
@bddbot go ahead and add to whitelist

**Comment by akbertram** [comment]
@bddbot please retest

**Comment by akbertram** [comment]
@bddbot retest this please

**Comment by akbertram** [comment]
Configured build to explicitly use JDK1.7 until soot dependency is upgraded. @bddbot retest this please

**Comment by akbertram** [comment]
retest this please

File: core/src/main/java/org/renjin/primitives/combine/LazyBuilder.java [hunk 1]
```diff
@@ -63,11 +63,11 @@ private Vector composeNames(String prefix, Vector vector) {
     Vector names = vector.getNames();
     int numElements = vector.length();
 
-    if(!Strings.isNullOrEmpty(prefix) || names != null) {
+    if(!Strings.isNullOrEmpty(prefix) || (names !=  Null.INSTANCE && names != null)) {
       hasNames = true;
     }
 
-    if(Strings.isNullOrEmpty(prefix) && names == null) {
+    if(Strings.isNullOrEmpty(prefix) && (names ==  Null.INSTANCE || names == null)) {
```
Comment (akbertram) [inline core/src/main/java/org/renjin/primitives/combine/LazyBuilder.java]: Would you mind dropping the `names != null` and `names == null` checks alltogether? `null` should be avoided alltogether in Renjin (don't know why I relapsed here) and I think it's best to keep that standard clear. 

You could add an `@Nonnull` to vector.getNames() to remove any ambiguity.

Comment (cwlaird3) [inline core/src/main/java/org/renjin/primitives/combine/LazyBuilder.java]: Sure - done.  Did not add the @NonNull check as it did not seem to be on the classpath ..

**Pull request merged**